### PR TITLE
Add Docs for Azure AI Studio Support for the Inference API

### PR DESCRIPTION
**Pull request by markjhoy**
Add docs for the Azure AI Studio embeddings and chat completion task type (from https://github.com/elastic/elasticsearch/pull/108472)

Also updates the [Semantic Search tutorial](https://www.elastic.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html) to add in Azure AI Studio as well.

### CODE REVIEW

File: docs/reference/inference/put-inference.asciidoc [hunk 1]
```diff
@@ -224,6 +226,59 @@ Your Azure OpenAI deployments can be found though the https://oai.azure.com/[Azu
...
+(Required, string)
+One of `token` or `realtime`.
+This specifies the type of endpoint that is used in your model deployment.
+There are https://learn.microsoft.com/en-us/azure/ai-studio/concepts/deployments-overview#billing-for-deploying-and-inferencing-llms-in-azure-ai-studio[two endpoint types available] for deployment through Azure AI Studio.
+"Pay as you go" endpoints are billed per token.
+For these you will need to specify `token` for your `endpoint_type`.
+For "real-time" endpoints which are billed per hour of usage, specify `realtime`.
+
+`rate_limit`:::
+(Optional, object) By default, the `azureaistudio` service sets a number of requests per minute rate limit at `240` requests per minute.
```
Comment (szabosteve) [inline docs/reference/inference/put-inference.asciidoc:274]:
```suggestion
(Optional, object) 
By default, the `azureaistudio` service sets a number of requests per minute rate limit at `240` requests per minute.
```

File: docs/reference/inference/put-inference.asciidoc [hunk 2]
```diff
@@ -224,6 +226,59 @@ Your Azure OpenAI deployments can be found though the https://oai.azure.com/[Azu
...
+This helps to minimize the number of rate limit errors returned from Azure AI Studio.
+If you wish to modify this, set the `requests_per_minute` setting of this object in your service settings:
```
Comment (szabosteve) [inline docs/reference/inference/put-inference.asciidoc:276]:
```suggestion
To modify this, set the `requests_per_minute` setting of this object in your service settings:
```

File: docs/reference/tab-widgets/inference-api/infer-api-search-widget.asciidoc [hunk 3]
```diff
@@ -67,6 +73,16 @@ include::infer-api-search.asciidoc[tag=openai]
 
 include::infer-api-search.asciidoc[tag=azure-openai]
 
+++++
```
Comment (szabosteve) [inline docs/reference/tab-widgets/inference-api/infer-api-search-widget.asciidoc:76]: The missing `</div>` causes the missing content issue in the preview.
```suggestion
++++
</div>
```

**Review by szabosteve — COMMENTED** [review]
Thank you for writing the docs for this service! I left a few comments; mostly nits to reduce word count and to harmonize phrasing. The only one that is slightly more important is the missing `</div>` that makes the search widget tab content disappear.

**Comment by markjhoy** [comment]
> Thank you for writing the docs for this service! I left a few comments; mostly nits to reduce word count and to harmonize phrasing. The only one that is slightly more important is the missing `</div>` that makes the search widget tab content disappear.

Thanks @szabosteve ! New commit is pushed addressing these.

**Review by szabosteve — APPROVED** [review]
Thanks again, LGTM! 🚢

**Pull request merged**